File: AiyaEffectsSDK/src/main/java/com/aiyaapp/camera/sdk/base/Log.java
/*
 *
 * Log.java
 * 
 * Created by Wuwang on 2016/11/8
 * Copyright © 2016年 深圳哎吖科技. All rights reserved.
 */
package com.aiyaapp.camera.sdk.base;

import com.aiyaapp.camera.sdk.AiyaEffects;

/**
 * Description:
 */
public class Log {

    private static final String tag="AiyaCameraEffect";
    private static boolean isDebug=true;

    private Log(){}

    public static void debug(boolean isDebug){
        Log.isDebug=isDebug;
        AiyaEffects.getInstance().set(ISdkManager.SET_LOG,isDebug?0x03:0x07);
    }

    public static void e(String info){
        if(isDebug){
            android.util.Log.e(tag,info);
        }
    }

    public static void e(String tag,String info){
        if (isDebug){
            android.util.Log.e(tag,info);
        }
    }

    public static void d(String info){
        if(isDebug){
            android.util.Log.d(tag,info);
        }
    }

    public static void d(String tag,String info){
        if(isDebug){
            android.util.Log.d(tag,info);
        }
    }


}

File: AiyaEffectSDK/AyCore/src/main/cpp/Log.cpp
//
// Created by aiya on 2017/7/15.
//

#include "Log.h"
#include "android/log.h"

#define log(a,b,c,d)                                        \
        if(a){                                              \
            va_list e;                                      \
            va_start(e ,d);                                 \
            __android_log_vprint(b,c,d,e);                  \
            va_end(e);                                      \
        }

static bool isDebug= true;
static char * mTag= (char *) "aiyaapp";

void Log::debug(bool debug) {
    isDebug=debug;
}

void Log::tag(char *tag) {
    mTag=tag;
}

void Log::e(const char *info, ...) {
    log(isDebug,ANDROID_LOG_ERROR,mTag,info);
}

void Log::i(const char *info, ...) {
    log(isDebug,ANDROID_LOG_INFO,mTag,info);
}

void Log::d(const char *info, ...) {
    log(isDebug,ANDROID_LOG_DEBUG,mTag,info);
}

void Log::te(char *tag, const char *info, ...) {
    log(isDebug,ANDROID_LOG_ERROR,tag,info);
}

void Log::ti(char *tag, const char *info, ...) {
    log(isDebug,ANDROID_LOG_INFO,tag,info);
}

void Log::td(char *tag, const char *info, ...) {
    log(isDebug,ANDROID_LOG_DEBUG,tag,info);
}